compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1776,7 +1784,18 @@ class Namer { typer: Typer =>
     if isConstructor then
       // set result type tree to unit, but take the current class as result type of the symbol
       typedAheadType(ddef.tpt, defn.UnitType)
-      wrapMethType(effectiveResultType(sym, paramSymss))
+      val res = wrapMethType(effectiveResultType(sym, paramSymss))
+      res
+/*      val unlink = sym.owner.is(JavaDefined) && sym.is(Synthetic) && sym.owner.derivesFrom(defn.JavaRecordClass) &&
+        sym.owner.unforcedDecls.lookupAll(sym.name).exists(c => c != sym && c.info.matches(res))
+
+      //println(s"sym: $sym, sym.owner ${sym.owner.flagsUNSAFE.flagsString}, sym ${sym.flagsUNSAFE.flagsString}")
+      //if sym.owner.is(JavaDefined) && sym.is(Synthetic) && sym.owner.derivesFrom(defn.JavaRecordClass) then
+      //  println(s"unlink: $unlink -- ${sym.owner.info.decl(sym.name).alternatives}")
+      if unlink then
+        sym.markAbsent()
+        NoType
+      else res*/
```

Comment:
Looks like some debug code that needs to be cleaned up.

Reply:
ooops that's right! thanks